qa/SidechainTestFramework/scutil.py

```diff
@@ -81,153 +88,284 @@ def sync_sc_mempools(api_connections, wait_for=25):
+                                   "com.horizen.ScBootstrappingTool",
+                                   "generatekey", json.dumps(jsonParameters)], stdout=subprocess.PIPE)
+        scBootstrapOutput = javaPs.communicate()[0]
+        secrets.append(json.loads(scBootstrapOutput))
+    return secrets
+
+
+# Maybe should we give the possibility to customize the configuration file by adding more fields ?
 
-#Maybe should we give the possibility to customize the configuration file by adding more fields ?
-def initialize_sc_datadir(dirname, n, genesisData):
+"""
+Create directories for each node and configuration files inside them.
+For each node put also genesis data in configuration files.
+Configuration data must be automatically generated and different from the ones generated for the other nodes.
+
+Parameters:
+ - dirname: directory name
+ - n: sidechain node nth
+ - account_secrets: a JSON array of secrets, in the following form:
+            [
+                {
+                    "secret":"first secret",
+                    "publicKey":"first public key"
+                },
+                {
+                    "secret":"second secret",
+                    "publicKey":"second public key"
+                },
+                ...,
+                {
+                    "secret":"nth secret",
+                    "publicKey":"nth public key"
+                }
+            ]
+ - genesis_info: a JSON object, to be included inside configuration file of the sidechain node nth, in the following form:
+             {
+                "scId": "id of the sidechain node",
+                "scGenesisBlockHex": "some value",
+                "powData": "some value",
+                "mcBlockHeight": xxx,
+                "mcNetwork": regtest|testnet|mainnet
+            }
+"""
+def initialize_sc_datadir(dirname, n, account_secrets=None, genesis_info=None):
     """Create directories for each node and configuration files inside them.
-       For each node put also genesis data in configuration files. 
-       Configuration data must be automatically generated and different from 
+       For each node put also genesis data in configuration files.
+       Configuration data must be automatically generated and different from
        the ones generated for the other nodes."""
-    genesis_secrets = {0 : "6882a61d8a23a9582c7c7e659466524880953fa25d983f29a8e3aa745ee6de5c0c97174767fd137f1cf2e37f2e48198a11a3de60c4a060211040d7159b769266", \
-                       1 : "905e2e581615ba0eff2bcd9fb666b4f6f6ed99ddd05208ae7918a25dc6ea6179c958724e7f4c44fd196d27f3384d2992a9c42485888862a20dcec670f3c08a4e", \
-                       2 : "80b9a06608fa5dbd11fb72d28b9df49f6ac69f0e951ca1d9e67abd404559606be9b36fb5ae7e74cc50603b161a5c31d26035f6a59e602294d9900740d6c4007f"}
+    jsonNode = generate_genesis_data(n, genesis_info) if genesis_info != None else {
```

Comment:
Please, use 2 different configuration templates:
1) with predefined genesis part (lines 202-215) and keep genesis data there, but not in py file
2) without predefined genesis part and put `genesis_info` from method parameters

Reply:
it can be even 2 different methods `initialize_sc_datadir`, check what is more convenient and minimalistic

Reply:
Got it. Taking in consideration that we have just a few tests implemented, we can remove this outdated stuff now and not to support it for future numerous tests. It will save us time. Just verify work-ability of that 2 SC tests

Reply:
ok

Reply:
maybe wen can leave 1 method `initialize_sc_datadir` and if both account_secrets!=None and genesis_info!=None then use template.conf, otherwise use template_predefined_genesis.conf

Reply:
Done. Added two methods `initialize_sc_datadir` and `initialize_default_sc_datadir`